### PR TITLE
Modified sync error log calls to include error type

### PR DESCRIPTION
**Pull request by chofnar**
Signed-off-by: Catalin Hofnar <catalin.hofnar@gmail.com>

**What type of PR is this?**

enhancement

**Which issue does this PR fix**:

#336 

**What does this PR do / Why do we need it**:

This PR adds a type evaluation function in utils.go so that we can also print the type of the error, not just its message. Also modified every log.error call to use this function.

**Does this PR introduce any user-facing change?**:

The user will now be able to see the type of the error in the logs.

Example:
From:
```
{"level":"error","error":"open /invalid/path/to/file: no such file or directory","goroutine":68,"caller":"/home/chofnar/go/src/github.com/project-zot/zot/pkg/extensions/sync/sync.go:543","time":"2022-04-15T17:45:33.8741831+03:00","message":"couldn't get registry credentials from /invalid/path/to/file"}
```
To:
```
{"level":"error","errorType":"*fs.PathError","error":"open /invalid/path/to/file: no such file or directory","goroutine":68,"caller":"/home/chofnar/go/src/github.com/project-zot/zot/pkg/extensions/sync/sync.go:543","time":"2022-04-15T17:45:33.8741831+03:00","message":"couldn't get registry credentials from /invalid/path/to/file"}
```

It is theoretically possible to change the "error" field name 
```
"error":"open /invalid/path/to/file: no such file or directory"
```
to 
```
"*fs.PathError":"open /invalid/path/to/file: no such file or directory"
```
by changing the zerolog ErrorFieldName. Unfortunately, this is not possible, as the value cannot be made to change dynamically for every error type - the variable is global and used by all loggers. A change would need to be implemented in zerolog to allow for this.

The solution in the PR simply adds a new field in the json message that zerolog outputs, which might come in handy if the users use software that monitors logs, such as Splunk.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/extensions/sync/on_demand.go [hunk 1]
```diff
@@ -98,7 +98,8 @@ func syncOneImage(imageChannel chan error, cfg Config, storeController storage.S
 
 		credentialsFile, err = getFileCredentials(cfg.CredentialsFile)
 		if err != nil {
-			log.Error().Err(err).Msgf("couldn't get registry credentials from %s", cfg.CredentialsFile)
+			log.Error().Str("errortype", TypeOf(err)).
```
Comment (rchincha) [inline pkg/extensions/sync/on_demand.go:101]: Can you show me an example output?

Comment (chofnar) [reply pkg/extensions/sync/on_demand.go:101]: This is what it looked like before
```
{"level":"error","error":"open /invalid/path/to/file: no such file or directory","goroutine":68,"caller":"/home/chofnar/go/src/github.com/project-zot/zot/pkg/extensions/sync/sync.go:543","time":"2022-04-15T17:45:33.8741831+03:00","message":"couldn't get registry credentials from /invalid/path/to/file"}
```
And this is what it looks like now
```
{"level":"error","errortype":"*fs.PathError","error":"open /invalid/path/to/file: no such file or directory","goroutine":68,"caller":"/home/chofnar/go/src/github.com/project-zot/zot/pkg/extensions/sync/sync.go:543","time":"2022-04-15T17:45:33.8741831+03:00","message":"couldn't get registry credentials from /invalid/path/to/file"}
```

**Comment by andaaron** [comment]
@chofnar please rebase.
@rchincha can we merge this?

File: pkg/extensions/sync/on_demand.go (hunk 1 above)
Comment (rchincha) [inline pkg/extensions/sync/on_demand.go:101]: Make this "errorType"

**Comment by rchincha** [comment]
@chofnar pls address comments, and rebase

File: pkg/extensions/sync/on_demand.go (hunk 1 above)
Comment (chofnar) [reply pkg/extensions/sync/on_demand.go]: Updated as requested

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**